### PR TITLE
Added E2E tests for legacy events

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Added E2E tests for legacy events
- NOTE: Event delivery verification part will be done in a separate PR.

**Related issue(s)**

- https://github.com/kyma-project/eventing-manager/issues/74

### CODE REVIEW

File: hack/e2e/common/testenvironment/test_environment.go [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+
+	// publish the event
+	if err := te.EventPublisher.SendLegacyEvent(eventSource, legacyEventType, payload); err != nil {
+		te.Logger.Debug(err.Error())
+		return err
+	}
+
+	// verify if the event was received by the sink.
+	te.Logger.Debug(eventID)
+	// TODO: implement me!
```
Comment (mfaizanse) [inline hack/e2e/common/testenvironment/test_environment.go:357]: It will be done in a separate PR.

File: hack/e2e/common/testenvironment/test_environment.go [hunk 2]
```diff
@@ -0,0 +1,377 @@
...
```
Comment (mfaizanse) [inline hack/e2e/common/testenvironment/test_environment.go:374]: It will be done in a separate PR.

File: hack/e2e/scripts/event_delivery_tests.sh [hunk 3]
```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+set -e
+
+# for our tests we need to port-forward the eventing-publisher-proxy.
+echo "Port-forwarding to the eventing-publisher-proxy using port: 38081"
+kubectl -n kyma-system port-forward svc/eventing-publisher-proxy 38081:80 &
+PID1=$!
+
+# This will will all the port-forwarding. We need this to be in a function so we can even call it, if our tests fails
```
Comment (friedrichwilken) [inline hack/e2e/scripts/event_delivery_tests.sh:9]:
```suggestion
# This will kill all the port-forwarding. We need this to be in a function so we can even call it, if our tests fails
```

Comment (friedrichwilken) [reply hack/e2e/scripts/event_delivery_tests.sh:9]: (This misspelling was initially done by me, I guess.)

**Review by friedrichwilken — CHANGES_REQUESTED** [review]

File: hack/e2e/common/testenvironment/test_environment.go [hunk 4]
```diff
@@ -0,0 +1,377 @@
...
+		panic(err)
+	}
+
+	testConfigs, err := env.GetE2EConfig()
+	if err != nil {
+		logger.Error(err.Error())
+		panic(err)
+
+	}
+	logger.Info(fmt.Sprintf("##### NOTE: Tests will run w.r.t. backend: %s", testConfigs.BackendType))
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply hack/e2e/common/testenvironment/test_environment.go:59]: `With Respect To`

File: hack/e2e/common/testenvironment/test_environment.go [hunk 5]
```diff
@@ -0,0 +1,377 @@
...
+	return common.Retry(Attempts, Interval, func() error {
+		// It's fine if the Namespace already exists.
+		return client.IgnoreAlreadyExists(te.K8sClient.Create(te.Context, fixtures.Namespace(te.TestConfigs.TestNamespace)))
+	})
+}
+
+func (te *TestEnvironment) DeleteTestNamespace() error {
+	return common.Retry(FewAttempts, Interval, func() error {
+		// It's fine if the Namespace already exists.
+		return client.IgnoreAlreadyExists(te.K8sClient.Delete(te.Context, fixtures.Namespace(te.TestConfigs.TestNamespace)))
```
[the remark this answers is not in the document]
Comment (mfaizanse) [reply hack/e2e/common/testenvironment/test_environment.go:86]: AFAIK, yes

File: hack/e2e/common/testenvironment/test_environment.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (friedrichwilken) [reply hack/e2e/common/testenvironment/test_environment.go:59]: thanks!

**Review by friedrichwilken — APPROVED** [review]
